From 32685c2a168659f002c2ed786a5202e4c7c1a05a Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 29 Apr 2022 06:36:56 +0300
Subject: [PATCH] [Sw-0] gaudi3: fix erom start address

erom in gaudi3 is executed on iccm . This patch modifies
the start address to use internal addressing.

This is required as:
-> Internal addressing would improve the execution speed compared
to cbu i/f addressing.
-> An exception is observed when using cbu i/f addressing when
instruction cache is enabled.

Change-Id: I2b8964c6be3ddf2d5c033b116fb1948c1a759a87
---

diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 0beb457..60c41fb 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -49,9 +49,9 @@
 EROM_TPM_TEXT_BASE=0x0300007FD3040000
 EROM_RECOVERY_TEXT_BASE=0x0300007FD3040000
 
-EROM_PCIE_START_ADDR=$EROM_PCIE_TEXT_BASE
-EROM_RECOVERY_START_ADDR=$EROM_RECOVERY_TEXT_BASE
-EROM_TPM_START_ADDR=$EROM_RECOVERY_TEXT_BASE
+EROM_PCIE_START_ADDR=0x0
+EROM_RECOVERY_START_ADDR=0x0
+EROM_TPM_START_ADDR=0x0
 
 IMAGE_TYPE_EROM=1
 IMAGE_TYPE_SPI=0
